# support for free_aligned_sized and free_sized were introduced in Heap_Layers 3-4 mar 2026:
#   - https://github.com/emeryberger/Heap-Layers/pull/72
#   - https://github.com/emeryberger/Heap-Layers/pull/71
#   - https://github.com/emeryberger/Heap-Layers/pull/70
# The introduction of free_aligened_sized and free_sized caused issues for Scalene 1.5.26.
# Last verified commit before these changes is:
#   - https://github.com/emeryberger/Heap-Layers/commit/89b9c6d0aa28773ccb187804b6f699824bdf7bb7
#
# author: Lara Peeters (UGent)
diff -ruN scalene-1.5.26.orig/GNUmakefile scalene-1.5.26/GNUmakefile
--- scalene-1.5.26.orig/GNUmakefile	2023-08-22 23:29:45.000000000 +0200
+++ scalene-1.5.26/GNUmakefile	2026-07-08 19:44:20.424408411 +0200
@@ -55,6 +55,7 @@
 
 vendor/Heap-Layers:
 	mkdir -p vendor && cd vendor && git clone https://github.com/emeryberger/Heap-Layers
+	cd vendor/Heap-Layers && git checkout 89b9c6d0aa28773ccb187804b6f699824bdf7bb7
 
 vendor/printf/printf.cpp:
 	mkdir -p vendor && cd vendor && git clone https://github.com/mpaland/printf
diff -ruN scalene-1.5.26.orig/Makefile scalene-1.5.26/Makefile
--- scalene-1.5.26.orig/Makefile	2023-08-22 23:29:45.000000000 +0200
+++ scalene-1.5.26/Makefile	2026-07-08 19:45:19.122031571 +0200
@@ -33,6 +33,7 @@
 
 vendor/Heap-Layers:
 	cd vendor && git clone https://github.com/emeryberger/Heap-Layers
+	cd vendor/Heap-Layers && git checkout 89b9c6d0aa28773ccb187804b6f699824bdf7bb7
 
 vendor/printf/printf.cpp:
 	cd vendor && git clone https://github.com/mpaland/printf
